src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -255,17 +255,38 @@ public E create(E entity, boolean flush) {
     /**
      * @return all entities for a particular type.
      */
-    public CandlepinQuery<E> listAll() {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria();
+    public List<E> listAll() {
+        EntityManager em = this.getEntityManager();
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<E> criteriaQuery = criteriaBuilder.createQuery(this.entityType);
+        Root<E> root = criteriaQuery.from(this.entityType);
+        criteriaQuery.select(root);
+
+        Predicate securityPredicate = this.getSecurityPredicate(this.entityType, criteriaBuilder, root);
+        if (securityPredicate != null) {
+            criteriaQuery.where(securityPredicate);
+        }
 
-        return this.cpQueryFactory.buildQuery(this.currentSession(), criteria);
+        return em.createQuery(criteriaQuery).getResultList();
     }
 
-    public CandlepinQuery<E> listAllByIds(Collection<? extends Serializable> ids) {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria()
-            .add(CPRestrictions.in("id", ids));
+    public List<E> listAllByIds(Collection<? extends Serializable> ids) {
+        EntityManager em = this.getEntityManager();
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<E> criteriaQuery = criteriaBuilder.createQuery(this.entityType);
+        Root<E> root = criteriaQuery.from(this.entityType);
+        criteriaQuery.select(root);
+
+        List<Predicate> predicates = new ArrayList<>();
+        predicates.add(root.get("id").in(ids));
```

Comment:
Note that this will run headlong into the IN-operator limit issues. The old code has the somewhat unreliable/buggy `CPRestrictions` variant, but that shouldn't be used anymore either. Instead we should be partitioning the input collection(s) as we do elsewhere. This can be done with `ParameterExpression`s and the same looping logic we use with our JPQL queries.

An example/tutorial of this can be found here: https://www.baeldung.com/jpa-query-parameters